subprojects/language-java/build.gradle.kts

```diff
@@ -85,7 +85,7 @@ strictCompile {
 
 packageCycles {
     // These public packages have classes that are tangled with the corresponding internal package.
-    excludePatterns.add("org/gradle/api/tasks/compile/**")
+    excludePatterns.add("org/gradle/api/tasks/**")
```

Comment:
More tangling? Can we somehow have less tangling?

Reply:
I don't see how we can. The new cycles are due to the executable utils classes for `Javadoc` and `JavaExec` that live in an internal package, but are used by the tasks themselves that are in the non-internal package.

Less tangling seems to be possible only if we replace tasks like `Javadoc` with an interface and have the implementation fully in the internal package.

Reply:
The solution to package cycles is often more packages. So we could introduce new `util` packages and put the new classes there. That would fix the cycles. Though it is probably not worth it. Your call.